code/game/objects/items/weapons/implants/implantaugment.dm

```diff
@@ -0,0 +1,198 @@
+//////////////////////////////
+//	Nanite Organ Implant
+//////////////////////////////
+/obj/item/weapon/implant/organ
+	name = "nanite fabrication implant"
+	desc = "A buzzing implant covered in a writhing layer of metal insects."
+	icon_state = "implant_evil"
+	origin_tech = list(TECH_MATERIAL = 5, TECH_BIO = 2, TECH_ILLEGAL = 2)
+
+	var/organ_to_implant = /obj/item/organ/internal/augment/bioaugment/thermalshades
+	var/organ_display_name = "unknown organ"
+
+/obj/item/weapon/implant/organ/get_data()
+	var/dat = {"
+<b>Implant Specifications:</b><BR>
+<b>Name:</b> \"GreyDoctor\" Class Nanite Hive<BR>
+<b>Life:</b> Activates upon implantation, destroying itself in the process.<BR>
+<b>Important Notes:</b> Nanites will fail to complete their task if a suitable location cannot be found for the organ.<BR>
+<HR>
+<b>Implant Details:</b><BR>
+<b>Function:</b> Nanites will fabricate: <span class='alien'>[organ_display_name]</span><BR>
+<b>Special Features:</b> Organ identification protocols.<BR>
+<b>Integrity:</b> N/A"}
+	return dat
+
+/obj/item/weapon/implant/organ/post_implant(var/mob/M)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+
+		var/obj/item/organ/NewOrgan = new organ_to_implant()
+
+		var/obj/item/organ/external/E = H.get_organ(NewOrgan.parent_organ)
+		to_chat(H, "<span class='notice'>You feel a tingling sensation in your [part].</span>")
+		if(E && !(H.internal_organs_by_name[NewOrgan.organ_tag]))
+			spawn(rand(1 SECONDS, 30 SECONDS))
+				to_chat(H, "<span class='alien'>You feel a pressure in your [E] as the tingling fades, the lump caused by the implant now gone.</span>")
+
+			NewOrgan.forceMove(H)
+			NewOrgan.owner = H
+			if(E.internal_organs == null)
+				E.internal_organs = list()
+			E.internal_organs |= NewOrgan
+			H.internal_organs_by_name[NewOrgan.organ_tag] = NewOrgan
+			H.internal_organs |= NewOrgan
+			NewOrgan.handle_organ_mod_special()
+
+			spawn(1)
+				if(!QDELETED(src))
+					qdel(src)
+
+		else
+			qdel(NewOrgan)
+			to_chat(H, "<span class='warning'>You feel a pinching sensation in your [part]. The implant remains.</span>")
+
+/obj/item/weapon/implant/organ/islegal()
+	return 0
+
+/*
+ * Arm / leg mounted augments.
+ */
+
+/obj/item/weapon/implant/organ/limbaugment
+	name = "nanite implant"
+
+	organ_to_implant = /obj/item/organ/internal/augment/armmounted/taser
+	organ_display_name = "physiological augment"
+
+	var/list/possible_targets = list(O_AUG_L_FOREARM, O_AUG_R_FOREARM)
+
+/obj/item/weapon/implant/organ/limbaugment/post_implant(var/mob/M)
```

Comment:
`setup_augment_slots(H, NewOrgan)` is used to select the target location for the implanted organ.

Reply:
Handling of NewOrgan is different, and it seemed safer to simply give it a unique post_implant rather than try to counteract any oddities from having the organ exist constantly in the implant.